src/base/install/getFilesInDir.m

```diff
@@ -0,0 +1,150 @@
+%                                            .gitignore). If the folder is
+%                                            not controlled by git, 'all'
+%                                            will be used. 
+%                                'untracked' - anything that is not ignored
+%                                              and not tracked. (new files)
+%                                'all'  - all files except for the git
+%                                         specific files (e.g. .git, .gitignore etc).                               
+%                                'COBRAIgnored' - use the COBRA Toolbox .gitignore
+%                                                 file to determine the ignored files. 
+%                                                 Slower than 'ignored',
+%                                                 since all files have to
+%                                                 be manually checked
+%                                                 against the expressions
+%                                                 specified in the
+%                                                 gitignored file, but
+%                                                 available on all folders
+%                                                 
+%                                (Default: 'all')   
+%                  restrictToPattern - give a regexp pattern to filter the
+%                                      files, this option is ignored if
+%                                      empty. (Default: '', i.e. ignored)
+%                  checkSubFolders - check the subfolders of the current
+%                                    directory. (Default: true)
+% 
+% EXAMPLES:
+%    Get all m files in the source folder:
+%    files = getFilesInDir('dirToList', [CBTDIR filesep 'src'], 'restrictToPattern', '\.m$');
+%    Get the git tracked files in the test Directory.
+%    files = getFilesInDir('dirToList', [CBTDIR filesep 'test'], 'gitTypeFlag', 'tracked');
+%    Get all git tracked files  which start with "MyFile" in the current directory
+%    files = getFilesInDir('gitTypeFlag', 'tracked', 'restrictToPattern', '^MyFile');
+%    Get only the gitIgnored files in the current folder
+%    files = getFilesInDir('gitTypeFlag', 'ignored');
+
+persistent COBRAIgnored
+
+if isempty(COBRAIgnored)
+    COBRAIgnored = regexptranslate('wildcard',getIgnoredFiles());
+end
+
+gitFileTypes = {'tracked','all','untracked','ignored','COBRAIgnored'};
+parser = inputParser();
+parser.addParamValue('dirToList',pwd,@(x) exist(x,'file') == 7);
+parser.addParamValue('gitTypeFlag','all',@(x) ischar(x) && any(strcmpi(x,gitFileTypes)));
+parser.addParamValue('restrictToPattern','',@(x) ischar(x));
+parser.addParamValue('checkSubFolders','',@(x) islogical(x) || (isnumeric(x) && x == 0 || x == 1));
+
+parser.parse(varargin{:});
+
+%get the absolute path of the files that are listed
+
+currentDir = cd(parser.Results.dirToList);
+absPath = pwd;
```

Comment:
What is `absPath`?

Reply:
Its the absolute Path to the specified folder.